include/net/lwm2m.h

```diff
@@ -150,6 +153,22 @@ struct lwm2m_ctx {
 	 */
 	bool use_dtls;
 
+#if defined(CONFIG_LWM2M_QUEUE_MODE_ENABLED)
+	/**
+	 * Flag to indicate that the socket is connection is suspended.
+	 * With queue mode, this will tell if there is a need to reconnect.
+	 *
+	 */
+	bool connection_suspended;
+
+	/**
+	 * Flag to indicate that the socket connection or registration is pending.
```

Comment:
```suggestion
	 * Flag to indicate that the client is buffering Notifications and Send messages. 
	 * This could be due to pending socket connection or pending registration.
```